server/etcdserver/corrupt_test.go

```diff
@@ -473,3 +478,42 @@ func (f *fakeHasher) TriggerCorruptAlarm(memberId types.ID) {
 	f.actions = append(f.actions, fmt.Sprintf("TriggerCorruptAlarm(%d)", memberId))
 	f.alarmTriggered = true
 }
+
+func TestClusterIdMismatch(t *testing.T) {
```

Comment:
This is a first test for hashKVHandler on http layer, can you add more test cases? for example a case that the request succeeds so we not only test a failure.